08_cell_line_prediction/plot_drug_response_regression.ipynb

```diff
@@ -0,0 +1,1285 @@
+{
```

Comment:
<p>Is there an issue with the way the data is displayed int the table? r^2 of -8.6 shouldn't be possible, but maybe there's a "e-1" somewhere that's getting truncated</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/pancancer-evaluation/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
I think a very negative r^2 value (even considerably more negative than -1) is possible for a really overfit model on a held-out dataset. [This article](https://towardsdatascience.com/explaining-negative-r-squared-17894ca26321) seems like a pretty decent explanation of how negative r^2 values can arise on test data.

To check this, I adapted the script in that article a bit to generate a dataset with more features and fewer train samples (i.e. to encourage overfitting). They generated 200 data points (160 train/40 test) with 10 features, and I changed this to 50 data points (40 train/10 test) with 50 features. Here are the r^2 values that gave me:

```
Train R2: 1.000, Test R2: -20.088
Train R2: 1.000, Test R2: -1.973
Train R2: 1.000, Test R2: -7.360
Train R2: 1.000, Test R2: -3.315
Train R2: 1.000, Test R2: -6.338
Train R2: 1.000, Test R2: -4.848
Train R2: 1.000, Test R2: -2.584
Train R2: 1.000, Test R2: -3.115
Train R2: 1.000, Test R2: -4.141
Train R2: 1.000, Test R2: -2.071
Train R2: 1.000, Test R2: -1.332
Train R2: 1.000, Test R2: -0.031
Train R2: 1.000, Test R2: -1.385
Train R2: 1.000, Test R2: -4.247
Train R2: 1.000, Test R2: -1.182
Train R2: 1.000, Test R2: -10.096
Train R2: 1.000, Test R2: -0.779
Train R2: 1.000, Test R2: -4.884
Train R2: 1.000, Test R2: -8.114
Train R2: 1.000, Test R2: -14.131
```
This is fairly similar to what I'm seeing in our regression results - the very negative r^2 values tend to happen for cancer types with relatively few cell lines (e.g. neuroblastoma, lymphoma) on the test data. The train r^2 values for those drugs/cancer types tend to be considerably higher (close to 1), so my guess is that they're just really overfit.

I'm not sure how to avoid this, exactly. We've tried filtering to cancer types that have more than a given number of cell lines in the dataset (I think I used 15 cell lines as the threshold) and it cut down on some of the variance in the results, but not that much. The general trend of the results is still the same, though, that there's little difference between feature selection methods and it's hard to beat the random selection baseline.